### PR TITLE
[RFR] Use redux-form onSubmit allowing remote submit

### LINKED ISSUE #1797: submitOnEnter not working
```submitOnEnter={false}``` does not work, form submits anyway.

**What you were expecting:**
Form not to submit.

**What happened instead:**
Form submits - acts like html form with ```method="get"``` - copies form parameter to url query for some reason.

**Steps to reproduce:**
create very standard component as in related code below

**Related code:**
```
export const UserEdit = (props) => (
    <Edit {...props}>
        <SimpleForm submitOnEnter={false}>
            <TextInput source="userName"/>
        </SimpleForm>
    </Edit>
);
```

**Other information:**

**Environment**

* React-admin version: v2.0.0-RC2
* Last version that did not exhibit the issue (if applicable): didn't checked
* React version: 16.3.2
* Browser: Chrome 66
* Stack trace (in case of a JS error): none

### PR DESCRIPTION
**Pull request by djhi**
Also fixes #1797

### CODE REVIEW

File: packages/ra-ui-materialui/src/button/SaveButton.js [hunk 1]
```diff
@@ -81,7 +82,7 @@ export class SaveButton extends Component {
                 className={classnames(classes.button, className)}
                 variant={variant}
                 type={type}
-                onClick={this.handleClick}
+                onClick={submitOnEnter ? undefined : this.handleClick}
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/button/SaveButton.js]: Does that mean that if submitOnEnter is true (which is the default), the save button doesn't work anymore? That sounds wrong.

File: packages/ra-ui-materialui/src/form/SimpleForm.spec.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ import TextInput from '../input/TextInput';
 describe('<SimpleForm />', () => {
     it('should embed a form with given component children', () => {
         const wrapper = shallow(
-            <SimpleForm>
+            <SimpleForm handleSubmit={() => {}}>
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/SimpleForm.spec.js]: If this is required, then your PR is a BC break

File: packages/ra-ui-materialui/src/button/SaveButton.js (hunk 1 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/button/SaveButton.js]: Nope, it fallback to the default html submit on enter, hence the necessity to have the `onSubmit` prop on the form

File: packages/ra-ui-materialui/src/form/SimpleForm.spec.js (hunk 2 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/form/SimpleForm.spec.js]: The `SimpleForm` in the tests is the not decorated by redux-form one. `handleSubmit` is injected by redux-form

File: packages/ra-ui-materialui/src/button/SaveButton.js (hunk 1 above)
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/button/SaveButton.js]: Gothcha

File: packages/ra-ui-materialui/src/form/SimpleForm.spec.js (hunk 2 above)
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/form/SimpleForm.spec.js]: ok then

**Comment by djhi** [comment]
> I'm lost in your PR between what's injected by redux-form and what's the developer's responsibility. A documentation on that feature would help.

Well actually nothing should have changed for our users. We just allow remote submit using the `submit` action for redux-form. I don't know what to document here

**Comment by djhi** [comment]
Ok, one thing has changed: we don't display a notification when clicking the submit button if the form is invalid. However, this notification was redundant with the error messages imo

**Comment by djhi** [comment]
Closed in favor of #2032

**Pull request closed without merging**